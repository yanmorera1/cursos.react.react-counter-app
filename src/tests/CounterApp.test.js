import { shallow } from "enzyme"
import React from 'react';
import '@testing-library/jest-dom';
import CounterApp from "../CounterApp";

describe('CounterApp tests', () => {
    let wrapper = shallow(
        <CounterApp />
    );

    beforeEach(() => {
        wrapper = shallow(
            <CounterApp />
        );
    })

    test('should show CounterApp successfully (do match w/ snapshot) and show default values', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('should show default value 100', () => {
        const valor = 100;
        const wrapper = shallow(
            <CounterApp value={valor} />
        );
        const text = parseInt(wrapper.find('h2').text().trim())
        expect(text).toBe(valor);
    })
    test('should be increment w/ button +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const text = parseInt(wrapper.find('h2').text().trim());
        expect(text).toBe(11);
    })
    test('should be decrement w/ button -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const text = parseInt(wrapper.find('h2').text().trim());
        expect(text).toBe(9);
    })
    test('should be reset value w/ button reset', () => {
        const wrapper = shallow(
            <CounterApp value={105} />
        );
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const text = parseInt(wrapper.find('h2').text().trim());
        
        expect(text).toBe(105);
    })
    
    
})
