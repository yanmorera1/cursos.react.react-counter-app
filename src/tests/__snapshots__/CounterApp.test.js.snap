// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp tests should show CounterApp successfully (do match w/ snapshot) and show default values 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
